frontend/public/components/utils/storage-class-dropdown.tsx

```diff
@@ -175,6 +162,30 @@ const StorageClassDropdownNoStorageClassOption = props => {
   </div>;
 };
 
+const getUnorderedItems = (items, filter) => {
```

Comment:
This is an untested snippet, but something like this should work. We shouldn't need to reduce, then filter. We can filter as we reduce the initial list of items.
```ts
const getUnorderedItems = (items, filter) =>
  items.reduce(
    (acc, resource) => ({
      ...acc,
      ...(filter &&
        filter(resource) && {
          [resource.metadata.name]: {
            kindLabel: 'StorageClass',
            name: resource.metadata.name,
            description: _.get(resource, 'metadata.annotations.description', ''),
            default: isDefaultClass(resource),
            accessMode: _.get(
              resource,
              ['metadata', 'annotations', 'storage.alpha.openshift.io/access-mode'],
              '',
            ),
            provisioner: resource.provisioner,
            type: _.get(resource, 'parameters.type', ''),
            zone: _.get(resource, 'parameters.zone', ''),
            resource,
          },
        }),
    }),
    {},
  );
```

Reply:
Also, the tests are failing because of a lint failure. This function needs to be declared before it is used.

Reply:
@TheRealJon Wouldn't this break code that doesn't use filter. In the sense if they don't have filter empty lists will start popping up.
```
const getUnorderedItems = (items, filter) => 
filter ?
  items.reduce( (acc, resource) => ({
    ...acc,
    ...(filter(resource) && {
      [resource.metadata.name]: {
        kindLabel: 'StorageClass',
        name: resource.metadata.name,
        description: _.get(resource, 'metadata.annotations.description', ''),
        default: isDefaultClass(resource),
        accessMode:  _.get(resource, ['metadata', 'annotations', 'storage.alpha.openshift.io/access-mode'], ''),
        provisioner: resource.provisioner,
        type: _.get(resource, 'parameters.type', ''),
        zone: _.get(resource, 'parameters.zone', ''),
        resource,
    }
  })
}),{})
  :
  items.reduce( (acc, resource) => ({
    ...acc,
      [resource.metadata.name]: {
        kindLabel: 'StorageClass',
        name: resource.metadata.name,
        description: _.get(resource, 'metadata.annotations.description', ''),
        default: isDefaultClass(resource),
        accessMode:  _.get(resource, ['metadata', 'annotations', 'storage.alpha.openshift.io/access-mode'], ''),
        provisioner: resource.provisioner,
        type: _.get(resource, 'parameters.type', ''),
        zone: _.get(resource, 'parameters.zone', ''),
        resource,
    }
  }),{});
```
How about this?